src/ape/managers/config.py

```diff
@@ -41,6 +44,11 @@ def __init__(self):
 
             self._plugin_configs[plugin_name] = config
 
+        # Top level config items
+        self.name = user_config.pop("name", "")
+        self.version = user_config.pop("version", "")
+        self.dependencies = user_config.pop("dependencies", [])
+
```

Comment:
Maybe this should move to above the plugin loop to prevent a malicious plugin from front-running these settings